plan/validator.go

```diff
@@ -34,12 +34,20 @@ func Validate(node ast.Node, inPrepare bool) error {
 	return v.err
 }
 
+// IsCacheable checkes whether or not the node is cacheable.
+func IsCacheable(node ast.Node, inPrepare bool) bool {
```

Comment:
should we also consider the UnCacheableFunctions ? I mean should we also call `Cacheable()` defined in `plan/cacheable_checker.go` ?

Reply:
1. It is possible. However, the existing Cacheable() only return true for select statements. Is there any issue for other statements ?

2. What is the difference between UnCacheableFunctions and UnFoldableFunctions ?